tools/CorrelationTestbed/Test-CorrelationInSandbox.ps1

```diff
@@ -0,0 +1,342 @@
+  [String] $ExePath,
+  [Parameter(HelpMessage = "Indicates that the local dev build should be used rather than the published package.")]
+  [Switch] $UseDev,
+  [Parameter(HelpMessage = "The directory where local dev build is located; only the release build works.")]
+  [String] $DevPackagePath,
+  [Parameter(HelpMessage = "The results output path.")]
+  [String] $ResultsPath,
+  [Parameter(HelpMessage = "The path to registry files that should be injected before the test.")]
+  [String] $RegFileDirectory
+)
+
+$ErrorActionPreference = "Stop"
+
+# Validate that the ExePath points to a reasonable location
+
+if (-not $ExePath)
+{
+  $ExePath = Join-Path $PSScriptRoot "InstallAndCheckCorrelation\x64\Release"
+}
+
+if (-not (Test-Path (Join-Path $ExePath "InstallAndCheckCorrelation.exe")))
+{
+  Write-Error -Category InvalidArgument -Message @"
+InstallAndCheckCorrelation.exe does not exist in the path $ExePath
+Either build it, or provide the location using -ExePath
+"@
+}
+
+# Validate that the local dev manifest exists
+
+if ($UseDev)
+{
+  if (-not $DevPackagePath)
+  {
+    $DevPackagePath = Join-Path $PSScriptRoot "..\..\src\AppInstallerCLIPackage\bin\x64\Release\AppX"
+  }
+
+  if ($DevPackagePath.ToLower().Contains("debug"))
+  {
+    Write-Error -Category InvalidArgument -Message @"
+The Debug dev package does not work for unknown reasons.
+Use the Release build or figure out how to make debug work and fix the scripts.
+"@
+  }
+  
+  if (-not (Test-Path (Join-Path $DevPackagePath "AppxManifest.xml")))
+  {
+    Write-Error -Category InvalidArgument -Message @"
+AppxManifest.xml does not exist in the path $DevPackagePath
+Either build the local dev package, or provide the location using -DevPackagePath
+"@
+  }
+}
+
+# Check if Windows Sandbox is enabled
+
+if (-Not (Get-Command 'WindowsSandbox' -ErrorAction SilentlyContinue)) {
+  Write-Error -Category NotInstalled -Message @'
+Windows Sandbox does not seem to be available. Check the following URL for prerequisites and further details:
+https://docs.microsoft.com/windows/security/threat-protection/windows-sandbox/windows-sandbox-overview
+
+You can run the following command in an elevated PowerShell for enabling Windows Sandbox:
+$ Enable-WindowsOptionalFeature -Online -FeatureName 'Containers-DisposableClientVM'
+'@
+}
+
+# Create output location for results
+
+if (-not $ResultsPath)
+{
+  $ResultsPath = Join-Path ([System.IO.Path]::GetTempPath()) (New-Guid)
+}
+
+if (Test-Path $ResultsPath)
+{
+  Remove-Item -Recurse $ResultsPath -Force
+}
+
+New-Item -ItemType Directory $ResultsPath > $nul
+
+# Close Windows Sandbox
+
+function Close-WindowsSandbox {
+    $sandbox = Get-Process 'WindowsSandboxClient' -ErrorAction SilentlyContinue
+    if ($sandbox) {
+      Write-Host '--> Closing Windows Sandbox'
+
+      $sandbox | Stop-Process
+      $sandbox | Wait-Process -Timeout 30
+
+      Start-Sleep 2
+
+      Write-Host
+    }
+    Remove-Variable sandbox
+}
+
+Close-WindowsSandbox

```

Comment:
Can you do this before line 85 where you remove the results folder?
If the script fails and the sandbox stays open, it will hold on to the results files and block removing the folder so we have to manually close the sandbox before running again.